### PR TITLE
feat(api-client):  cancel request and stop streaming responses

### PR DESCRIPTION
**Pull request by DemonHa**
This PR restores the Cancel Request functionality and introduces a Stop Streaming action while a streaming response is in progress.

It also ensures streaming responses are properly terminated when the user navigates away or when a new request/response cycle starts, preventing orphaned streams from continuing in the background and improving overall request lifecycle management.

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset. 
- [x] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> Restores request cancellation and adds robust streaming controls across the API client.
> 
> - Centralizes lifecycle via `WorkspaceEventBus`: `OperationBlock` stores `AbortController`, emits `hooks:on:request:sent/complete`, cancels on unmount/route change, and stops previous `response.reader` before new requests
> - Streaming UI: `ResponseBodyStreaming` adds a "Cancel" action, prevents race conditions, properly resets `TextDecoder`, and cleans up readers; error handling and auto-scroll improved
> - Response view refactor: `ResponseBlock`, `ResponseMetaInformation`, and `ResponseLoadingOverlay` switch from store events to `eventBus`; loading overlay and stopwatch tied to hook events; `ResponseEmpty` actions routed through `eventBus`
> - Address bar: loading animation exposes `isLoading`; disables "Send" while requesting
> - Extensive tests added for streaming, cancel behavior, event wiring, and decoder edge cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 4d2bcd281300a17017fa7535d968bb3cd3fff420. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/api-client/src/v2/blocks/response-block/components/ResponseEmpty.vue [hunk 1]
```diff
@@ -1,39 +1,25 @@
...
-const { totalPerformedRequests, layout, appVersion, events } = defineProps<{
+const { totalPerformedRequests, layout, appVersion } = defineProps<{
   /** Client layout */
   layout: ClientLayout
   /** Total number of performed requests */
   totalPerformedRequests: number
   /** Application version */
   appVersion: string
-  /** Event bus */
-  events: ReturnType<typeof createStoreEvents>
```
[the remark this answers is not in the document]
Comment (DemonHa) [reply packages/api-client/src/v2/blocks/response-block/components/ResponseEmpty.vue:20]: We already handle those globally
